cmake/modules/OmrTargetSupport.cmake

```diff
@@ -75,13 +75,25 @@ function(omr_add_library name)
 
 	if(NOT opt_NOWARNINGS AND NOT lib_type STREQUAL "INTERFACE")
 		if(OMR_WARNINGS_AS_ERRORS)
-			target_compile_options(${name} PRIVATE ${OMR_WARNING_AS_ERROR_FLAG})
+			target_compile_options(${name}
+				PRIVATE
+					$<$<COMPILE_LANGUAGE:C>:${OMR_WARNING_AS_ERROR_FLAG}>
+					$<$<COMPILE_LANGUAGE:CXX>:${OMR_WARNING_AS_ERROR_FLAG}>
```

Comment:
My concern here is that we lose warnings as errors for all non c/c++ languages, not just `ASM_NASM` eg `ASM` and `ASM_MASM` I think a better solution would be to make `OMR_<LANG>_WARNINGS_AS_ERROR_FLAG` variables and iterate the `ENABLED_LANGUAGES` property to apply all the flags

Reply:
@dnakamura would you be ok if we left this as is and if I opened an issue to track adding `OMR_<LANG>_WARNINGS_AS_ERROR_FLAG` which I will work on immediately after this PR is merged? Reason I'm suggesting this is due to the size of the PR and we just got the builds fully green so I want to merge what we have to enable the warnings as errors to prevent additional warnings being introduced. If you can give me a thumbs up I will open up an issue, assign to myself, and start working on it in the background as this PR gets reviewed.

Reply:
I'm gonna go ahead with the coordinated merge. @fjeremic I think just open up the issue and fix this limitation when you can.

Reply:
Opened #6053 to track this.